internal/sirius/my_details.go

```diff
@@ -45,9 +49,47 @@ func (c *Client) MyDetails(ctx context.Context, cookies []*http.Cookie) (MyDetai
 	}
 
 	if resp.StatusCode != http.StatusOK {
-		return v, errors.New("returned non-2XX response")
+		return v, errors.New("returned non-2XX response: " + strconv.Itoa(resp.StatusCode))
 	}
 
 	err = json.NewDecoder(resp.Body).Decode(&v)
 	return v, err
 }
+
+func (c *Client) EditMyDetails(ctx context.Context, cookies []*http.Cookie, id int, phoneNumber string) (ValidationErrors, error) {
+	var v struct {
+		Status           int              `json:"status"`
+		Detail           string           `json:"detail"`
+		ValidationErrors ValidationErrors `json:"validation_errors"`
+	}
+
+	var body = strings.NewReader("{\"phoneNumber\":\"" + phoneNumber + "\"}")
+
+	req, err := c.newRequest(
+		ctx,
+		http.MethodPut,
+		"/api/v1/users/"+strconv.Itoa(id)+"/updateTelephoneNumber",
+		body,
+		cookies,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := c.http.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode == http.StatusUnauthorized {
+		return nil, ErrUnauthorized
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		err = json.NewDecoder(resp.Body).Decode(&v)
+		return v.ValidationErrors, err
```

Comment:
We probably never care about a JSON decoding error, the status code will tell us more, so maybe something like?

```go
err = json.NewDecoder(resp.Body).Decode(&v)
if err == nil {
  return v.ValidationErrors
} 

return fmt.Errorf("returned non-2XX response: %d", resp.StatusCode)
```

Or check for `err == io.EOF`

```go
err = json.NewDecoder(resp.Body).Decode(&v)
if err == nil {
  return v.ValidationErrors
} 
if err == io.EOF {
  return fmt.Errorf("returned non-2XX response: %d", resp.StatusCode)
}
return err
```

?